clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -2596,6 +2596,7 @@ static void ParseLangArgs(LangOptions &Opts, ArgList &Args, InputKind IK,
       }
     }
     Opts.SYCLExplicitSIMD = Args.hasArg(options::OPT_fsycl_esimd);
+    Opts.EnableDAEInSpirKernels = Args.hasArg(options::OPT_fenable_sycl_dae);
```

Comment:
Ideally hasFlag should be used for things that can be enabled/disabled. Added 'flags' should typically have a 'no' version to so we can disable it easily in cmake/make/etc.  hasArg is for arguments that are not flags. E.g -triple xyz.

That said, the codebase does a bad job following these rules but maybe we can do it the right way when we catch it

Reply:
Ok, I see. This is just a temporary flag until other pieces will be in place. @elizabethandrews, are you OK with keeping it as it is for the time being?

Reply:
Oh I see. This flag will be removed once full implementation is done?  If so, I guess it is alright to keep it this way for now. If you however plan on keeping this flag to control this optimization, I would prefer just changing it now.

Reply:
I think the current plan is to keep this optimization always enabled once the other components will be in place.

Reply:
Alright. Thanks for clarifying. As long as this is eventually removed, I'm ok.

Reply:
Elizabeth has a point. Now we need to revert changes in 4 different files to remove the option when feature implementation is complete. 
Probably it would easier to have something like this in BackendUtils.cpp
```c++
  if (LangOpts.SYCLIsDevice && !CodeGenOpts.DisableLLVMPasses &&
      !LangOpts.SYCLExplicitSIMD && false /*LangOpts.EnableDAEInSpirKernels*/) // remove false once feature is ready
    PerModulePasses.add(createDeadArgEliminationSYCLPass());
```
We can try it out next time.